packages/orchestrator-ui-components/src/components/WFOTable/WFOBasicTable/WFOTableHeaderCell.tsx

```diff
@@ -1,32 +1,112 @@
-import React, { FC, ReactNode } from 'react';
+import React, { FC, useState } from 'react';
 
 import { SortOrder } from '../../../types';
 import { WFOSortDirectionIcon } from './WFOSortDirectionIcon';
+import {
+    EuiFieldSearch,
+    EuiHorizontalRule,
+    EuiPopover,
+    EuiText,
+    useGeneratedHtmlId,
+} from '@elastic/eui';
+import { useWithOrchestratorTheme } from '../../../hooks';
+import { WFOSortButtons } from '../WFOSortButtons';
+import { getStyles } from './styles';
 
 export type WFOTableHeaderCellProps = {
-    sortDirection?: SortOrder;
-    onClick?: () => void;
+    fieldName: string;
+    sortOrder?: SortOrder;
     isSortable?: boolean;
-    children: ReactNode;
+    onSetSortOrder?: (updatedSortOrder: SortOrder) => void;
+    onSearch?: (searchText: string) => void;
+    children: string;
 };
 
 export const WFOTableHeaderCell: FC<WFOTableHeaderCellProps> = ({
-    sortDirection,
+    fieldName,
+    sortOrder,
     children,
-    onClick,
     isSortable = true,
-}) => (
-    <div
-        css={{
-            display: 'flex',
-            alignItems: 'center',
-            cursor: isSortable ? 'pointer' : 'not-allowed',
-        }}
-        onClick={isSortable ? onClick : undefined}
-    >
-        <div>{children}</div>
-        {isSortable && sortDirection ? (
-            <WFOSortDirectionIcon sortDirection={sortDirection} />
-        ) : null}
-    </div>
-);
+    onSetSortOrder,
+    onSearch,
+}) => {
+    const {
+        headerCellContentStyle,
+        headerCellPopoverHeaderStyle,
+        headerCellPopoverHeaderTitleStyle,
+        headerCellPopoverContentStyle,
+        getHeaderCellButtonStyle,
+    } = useWithOrchestratorTheme(getStyles);
+
+    const smallContextMenuPopoverId = useGeneratedHtmlId({
+        prefix: 'smallContextMenuPopover',
+    });
+
+    const [isPopoverOpen, setPopover] = useState(false);
+    const handleButtonClick = () => setPopover(!isPopoverOpen);
+    const closePopover = () => setPopover(false);
+
+    const handleChangeSortOrder = (updatedSortOrder: SortOrder) => {
+        if (onSetSortOrder) {
+            onSetSortOrder(updatedSortOrder);
+            closePopover();
```

Comment:
Does it make sense to move the closePopover outside of this if? Shouldn't it always close if this function, also on the off chance onSetSortOrder isn't set?

Reply:
I would say no, because you would give the user the impression that something is happening/changed.